packages/components/src/TitleBlockZen/types.ts

```diff
@@ -0,0 +1,209 @@
+  navigationTabs?: NavigationTabs
+  collapseNavigationAreaWhenPossible?: boolean
+  textDirection?: TextDirection
+  surveyStatus?: SurveyStatus
+  titleAutomationId?: string
+  breadcrumbAutomationId?: string
+  breadcrumbTextAutomationId?: string
+  avatarAutomationId?: string
+  subtitleAutomationId?: string
+  sectionTitleAutomationId?: string
+  sectionTitleDescriptionAutomationId?: string
+}
+
+/**
+ * ### PrimaryActionProps
+ *
+ * The primary action (the "main" button in the top right) can either be a Button,
+ * or a Button that reveals a Menu (a menu button).
+ *
+ * For a button, pass in an object containing the same props you would normally use
+ * for a Button component.
+ *
+ * For a menu button, pass in a `MenuGroup`:
+ * ```typescript
+ * {
+ *   label: string
+ *   menuItems: MenuItemProps[]
+ * }
+ * ```
+ * Using the `label`, the Title Block will render a Button with a chevron icon and your `menuItems` will appear
+ * in the dropdown menu when you click it. (`MenuItemProps` is a type imported from the `Menu` component.)
+ */
+
+export type PrimaryActionProps =
+  | (MenuGroup & { badge?: TitleBlockBadgeProps })
+  | ((TitleBlockButtonProps | TitleBlockCustomButtonProps) & {
+      badge?: TitleBlockBadgeProps
+    })
+
+/**
+ * ### SecondaryActionsProps
+ *
+ * Secondary actions can only be buttons or button menus
+ * (a button that reveals a dropdown menu).
+ *
+ * **IMPORTANT:** The visual order of these from left to right should always be:
+ * ```text
+ * buttons -> menu buttons
+ * ```
+ *
+ * For a button, pass in an object containing the same props you would normally use
+ * for a Button component.
+ *
+ * For a menu button, pass in a `MenuGroup`:
+ *
+ * ```typescript
+ * {
+ *   label: string
+ *   menuItems: MenuItemProps[]
+ * }
+ * ```
+ * (`MenuItemProps` is imported from the Menu component.)
+ *
+ */
+export type SecondaryActionsProps = SecondaryActionItemProps[]
+
+export type SecondaryActionItemProps =
+  | MenuGroup
+  | (
+      | ButtonWithHrefNotOnClick
+      | ButtonWithOnClickNotHref
+      | TitleBlockCustomButtonProps
+    )
+
+export type TitleBlockBadgeProps = {
+  text: string
+  animateChange?: boolean
+}
+
+export type TitleBlockButtonProps = TitleBlockDistributiveOmit<
+  ButtonProps,
+  "onClick" | "component"
+> & {
+  onClick?: (e: any) => void
+}
+
+export type TitleBlockCustomButtonProps = TitleBlockDistributiveOmit<
+  TitleBlockButtonProps,
+  "component"
+> & {
+  className?: string
+  component: (props: CustomButtonProps) => JSX.Element
+}
+
+export type TitleBlockMenuItemProps =
+  | (Omit<MenuItemProps, "action"> & {
+      action: ((e: any) => void) | string
+    })
+  | TitleBlockCustomButtonProps
+
+export type ButtonWithHrefNotOnClick = TitleBlockDistributiveOmit<
+  ButtonProps,
+  "onClick"
+>
+export type ButtonWithOnClickNotHref = TitleBlockDistributiveOmit<
+  TitleBlockButtonProps,
+  "href"
+>
+
+export type MenuGroup = {
```

Comment:
Fingers crossed this doesn't have any conflict with Menu types 🤞🏻 
Name spacing would be safest though 😉 (But I can see a lot of types without name spaces, so not sure if it would break too much, thus optional-ish for now)